src/app/products/product-list/product-list.component.ts

```diff
@@ -1,14 +1,22 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
 import { MatBottomSheet } from '@angular/material/bottom-sheet';
 import { ColDef, GridOptions, RowDoubleClickedEvent } from 'ag-grid-community';
-import { filter, switchMap } from 'rxjs';
+import { filter, interval, Subject, switchMap, takeUntil } from 'rxjs';
 import { ProductDetailComponent } from '../product-detail/product-detail.component';
 import { Product } from '../product-http.service';
 import { ProductService } from '../product.service';
 
+const SECOND = 1000;
+const MINUTE = SECOND * 60;
+
 @Component({
   selector: 'y42-product-list',
-  template: `<ag-grid-angular
+  template: `
```

Comment:
Template should be moved to a html file as well. But from my point of view that is not required, when the file is not bigger than 400 lines of code